scripts/target-platform

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env bash
+
+: "${TARGETPLATFORM=}"
+: "${TARGETOS=}"
+: "${TARGETARCH=}"
+: "${TARGETVARIANT=}"
+
+# get TARGETOS/TARGETARCH/TARGETVARIANT from TARGETPLATFORM
+if [ -n "$TARGETPLATFORM" ]; then
+  os="$(echo $TARGETPLATFORM | cut -d"/" -f1)"
+  arch="$(echo $TARGETPLATFORM | cut -d"/" -f2)"
+  if [ -n "$os" ] && [ -n "$arch" ]; then
+    TARGETOS="$os"
+    TARGETARCH="$arch"
+    case "$arch" in
+      "arm")
+        case "$(echo $TARGETPLATFORM | cut -d"/" -f3)" in
+          "v5")
+            TARGETVARIANT="v5"
+            ;;
+          "v6")
+            TARGETVARIANT="v6"
+            ;;
+          "v8")
+            TARGETVARIANT="v8"
+            ;;
+          *)
+            TARGETVARIANT="v7"
+            ;;
+        esac
+        ;;
+      "mips"*)
+        TARGETVARIANT="$(echo $TARGETPLATFORM | cut -d"/" -f3)"
+        ;;
+    esac
+  fi
+fi
+
+# current arch/variant
+CUROS="linux"
+case "$(uname -m)" in
```

Comment:
didn't `xx` also have a utility for something like this? (wondering if we can somehow centralise this effort)

Reply:
yes there is the `xx-info` and we should use it in a follow-up when we will be able to be sandboxed inside a Dockerfile. See https://github.com/docker/docker-ce-packaging/pull/665#discussion_r838482622